gradio/interface.py

```diff
@@ -706,8 +706,7 @@ def attach_interpretation_events(
             interpretation_btn.click(
                 self.interpret_func,
                 inputs=self.input_components + self.output_components,
-                outputs=interpretation_set
-                or [] + [input_component_column, interpret_component_column],  # type: ignore
+                outputs=interpretation_set + [input_component_column, interpret_component_column],  # type: ignore
```

Comment:
What about this (which was the change I intended to make when originally refactoring this code):

```suggestion
                outputs=(interpretation_set or [])+ [input_component_column, interpret_component_column],  # type: ignore
```

The reason being that `interpretation_set` can actually be `None` (if the Interface type is OUTPUT_ONLY).

Reply:
Sounds good @abidlabs ! Thanks for the suggestion!

Reply:
Yes I was just thinking about this. So it seems like for outputs, we can now take any `Block` or `List[Block]`, because any `Block` could have properties that can be updated. This is going to complicate typing though because Python doesn't like it when you pass in a `List[Component]` to something that takes in a `List[Block]` even if `Component` is a subclass of `Block`, see https://github.com/microsoft/pyright/issues/130